src/libs/defaultRedirection/defaultRedirection.ts

```diff
@@ -0,0 +1,131 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+import CozyClient, { deconstructCozyWebLinkWithSlug } from 'cozy-client'
+
+import { formatRedirectLink } from '/libs/functions/formatRedirectLink'
+import { logger } from '/libs/functions/logger'
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+import strings from '/constants/strings.json'
+
+export const log = logger('AppBundle')
+
+export const NAVIGATION_APP_SLUG = 'home'
+
+export const DEFAULT_REDIRECTION_DELAY_IN_MS = 3000
+
+export interface SettingsCollection {
+  get: (path: string) => Promise<InstanceSettings>
+}
+
+interface InstanceSettings {
+  data: {
+    attributes: {
+      default_redirection: string
+    }
+  }
+}
+
+export const isDefaultRedirectionUrlNavigationApp = (
+  defaultRedirectionUrl: string,
+  client: CozyClient
+): boolean => {
+  const subdomainType = client.capabilities.flat_subdomains ? 'flat' : 'nested'
+
+  const { slug } = deconstructCozyWebLinkWithSlug(
+    defaultRedirectionUrl,
+    subdomainType
+  )
+
+  return slug === NAVIGATION_APP_SLUG
+}
+
+export const fetchDefaultRedirectionUrl = async (
+  client: CozyClient
+): Promise<string | undefined> => {
+  let defaultRedirectionUrl
+
+  try {
+    const settingsCollection = client.collection(
+      'io.cozy.settings'
+    ) as SettingsCollection
```

Comment:
You should almost never use the `collection` directly. `client.query()` always. 

`client.query(Q('io.cozy.settings').getById('instance'))` should do the trick ;)

Reply:
Also, client.query should always have a fectchPolicy and be named. By default, the getById will be named automatically by cozy-client itself, but it's a good habit to do it: 
`as: io.cozy.settings/instance`

Also the settings of an instance should not be often changed, so you can use a long policy ;)

Reply:
I am not sure about the long policy because this request will be used :
- the first time so we need fresh data
- when we open the app 10s later to update the value if the user updated it. So with a long policy the user will need to wait before seeing the change even after an app reload

Reply:
> the first time so we need fresh data

The first time, you'll not have fetched data, so the fetch policy will ne be applied. 

> when we open the app 10s later to update the value if the user updated it. So with a long policy the user will need to wait before seeing the change even after an app reload

Oh you right, I forgot about `to update the value if the user updated it` the fact that the user will be able to update the value.